cubi_tk/sodar/pull_data_collection.py

```diff
@@ -0,0 +1,367 @@
+        )
+
+        # Retrieve files from iRODS
+        iRODSTransfer(transfer_jobs).get(self.args.overwrite)
+
+        logger.info("All done. Have a nice day!")
+        return 0
+
+    @staticmethod
+    def parse_sample_tsv(tsv_path, sample_col=1, n_header_cols=1, skip_comments=True) -> List[str]:
+        samples = []
+        with open(tsv_path, "r") as f:
+            for _ in range(n_header_cols):
+                f.readline()
+            for line in f:
+                if not line.strip():
+                    continue
+                if line.startswith("#") and skip_comments:
+                    continue
+                try:
+                    samples.append(line.split("\t")[sample_col - 1])
+                except IndexError:
+                    logger.error(
+                        f"Error extracting column no. {sample_col} from {tsv_path}."
+                        f"Occured in line: {line}"
+                    )
+                    raise
+
+        return samples
+
+    @staticmethod
+    def filter_irods_file_list(
+        remote_files_dict: Dict[str, List[iRODSDataObject]],
+        common_assay_path: str,
+        file_patterns: List[str],
+        samples: List[str],
+        substring_match: bool = False,
+    ) -> Dict[str, List[iRODSDataObject]]:
+        """Filter iRODS collection based on identifiers (sample id or library name) and file type/extension.
+
+        :param remote_files_dict: Dictionary with iRODS collection information. Key: file name as string (e.g.,
+        'P001-N1-DNA1-WES1.vcf.gz'); Value: iRODS data (``iRODSDataObject``).
+        :type remote_files_dict: dict
+
+        :param common_assay_path: Path common to all files. If provided, files in this path will be stripped.
+        :type common_assay_path: str
+
+        :param file_patterns: List of file patterns to use for file selection. Ignored if empty.
+        :type file_patterns: list of strings
+
+        :param samples: List of collection identifiers or substrings. Ignored if empty.
+        :type samples: list
+
+        :param substring_match: Fiter by extact collection matches or by substring matches.
+        :type substring_match: bool
+
+        :return: Returns dictionary: Key: sample (collection name [str]); Value: list of iRODS objects.
+        """
+        # Initialise variables
+        filtered_dict = defaultdict(list)
+
+        # Iterate
+        for filename, irodsobjs in remote_files_dict.items():
+            for irodsobj in irodsobjs:
+                # Path needs to be stripped down to collections (=remove assay part & upwards)
+                path = PurePosixPath(irodsobj.path).relative_to(PurePosixPath(common_assay_path))
+
+                collection = path.parts[0]
+
+                # Check if collection (=1st element of striped path) matches any of the samples
+                if samples and not substring_match:
+                    sample_match = any([s.lower() == collection.lower() for s in samples])
```

Comment:
why are matches case insensitive? is irods actually case insensitive?

Reply:
mmh, I thought I'd taken that from somewhere else in cubi-tk but that doesn't seem to be the case. We can go case-sensitive I guess